src/routes/adminPanel/adminPanelTable/adminPanelTable.component.tsx

```diff
@@ -0,0 +1,48 @@
+import { useEffect, useState } from 'react';
+import { ExtendedRadarTechnology, TechnologyTable } from '../adminPanel.types';
+import { Styles } from './adminPanelTable.styles';
+import { Table } from './table.component';
+
+interface AdminPanelTableProps {
+  columns: TechnologyTable[];
+  rows: ExtendedRadarTechnology[];
+}
+
+export const AdminPanelTable = ({ columns, rows }: AdminPanelTableProps) => {
+  const [data, setData] = useState<ExtendedRadarTechnology[]>([]);
+  const [skipPageReset, setSkipPageReset] = useState(false);
+
+  useEffect(() => {
+    setData(rows);
+  }, [rows]);
+
+  useEffect(() => {
+    setSkipPageReset(false);
+  }, [data]);
+
+  const updateMyData = (rowIndex: number, columnId: string, value: string | number | boolean) => {
+    setSkipPageReset(true);
```

Comment:
Could you describe what is this state doing, and why it's immediately changed on the next render with useEffect?

Reply:
Here is more explanation about this: https://react-table-v7.tanstack.com/docs/examples/editable-data